src/main/java/touro/GameOfLife/Grid.java

```diff
@@ -1,4 +1,77 @@
 package touro.GameOfLife;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
 public class Grid {
+    private Direction[] directions = Direction.values();
+
+    //TODO: find a reasonable height and width
+    public static final int HEIGHT = 20;
+    public static final int WIDTH = 20;
+
+    private final List<Cell> allCells = new ArrayList<>();
```

Comment:
Is a List the best way to keep your cells?

The problem with just taking what we did in Snake is that what was good for that problem might not be the best solution here.

Reply:
I do understand that it's not always best to copy what we did in another project. 
In this case, I could not think of a better way to hold all the cells so that we can later iterate over them to count their live neighbors, but am open to suggestions

Reply:
let's say we are at cell 5,5
and we want to know if cell 5,6 is alive.
What's the easiest data structure we can use to find that out _quickly_?

Reply:
Our group had a meeting, and updated our strategy which should hopefully be faster. Instead of checking each cell for it's neighbors, the code now looks at look the live cells, and makes a hashmap of who it is a neighbor of. Once that hashmap is made, the we then iterate over it to see which cells should be included after the grid advances.

Reply:
speed-wise, this means for every `advance()`, the `liveCells` are iterated once, and the `potentialLiveCells` are iterated once. and all lookups for hashes are O(1)